ideascube/configuration/management/commands/config.py

```diff
@@ -0,0 +1,153 @@
+
+        list = subs.add_parser(
+            'list', help='List configuration namespaces and keys')
+        list.add_argument(
+            'namespace', nargs='?',
+            help='Only list configuration keys for this namespace')
+        list.set_defaults(func=self.list_configs)
+
+        reset = subs.add_parser(
+            'reset', help='Reset a configuration option to its default value')
+        reset.add_argument('namespace', help='The configuration namespace')
+        reset.add_argument('key', help='The configuration key')
+        reset.set_defaults(func=self.reset_config)
+
+        set = subs.add_parser(
+            'set', help='Set the value of a configuration option')
+        set.add_argument('namespace', help='The configuration namespace')
+        set.add_argument('key', help='The configuration key')
+        set.add_argument('value', help='The new value')
+        set.set_defaults(func=self.set_config)
+
+        self.parser = parser
+
+    def handle(self, *_, **options):
+        if 'func' not in options:
+            self.parser.print_help()
+            self.parser.exit(1)
+
+        options['func'](options)
+
+    def _evaluate_value(self, value):
+        if value in ('true', 'false'):
+            # Special case booleans for a better UX
+            value = value.capitalize()
+
+        try:
+            return ast.literal_eval(value)
+
+        except (SyntaxError, ValueError):
+            # Admin probably wanted a string and didn't use enough quotes
+            # examples:
+            #     $ ideascube config set server site-name thename
+            #     $ ideascube config set server site-name 'the name'
+            return str(value)
+
+    def get_config(self, options):
+        namespace = options['namespace']
+        key = options['key']
+
+        try:
+            value = get_config(namespace, key)
+
+        except (NoSuchConfigurationNamespaceError,
+                NoSuchConfigurationKeyError) as e:
+            raise CommandError(e)
+
+        print('%r' % value)
+
+    def help_for_config(self, options):
+        data = {
+            'namespace': options['namespace'],
+            'key': options['key'],
+        }
+
+        try:
+            data.update(get_config_data(data['namespace'], data['key']))
+
+        except (NoSuchConfigurationNamespaceError,
+                NoSuchConfigurationKeyError) as e:
+            raise CommandError(e)
+
+        print(
+            '# {namespace} {key}\n\n'
+            '{summary}\n\n'
+            '* type: {pretty_type}\n'
+            '* default value: {default!r}'.format(**data))
+
+    def list_configs(self, options):
+        namespace = options['namespace']
+
+        if namespace is None:
+            for namespace in get_all_namespaces():
+                print(namespace)
+
+            return
+
+        try:
+            for key in get_namespaced_configs(namespace):
+                print("%s %s" % (namespace, key))
```

Comment:
> Now I'm certainly not saying the UX I designed is perfect, far from it. But it is a global, consistent design based on certain assumptions, and if we're going to change it, I'd like us to instead challenge these very assumptions with sufficient feedback from real-world usage, and then evolve a new global, consistent design.

I understand that, and also that my suggestion is not fully aligned with the current design. I actually think the design is good.
But what I mean is that I've been trying to use the command, and the first thing I've wanted to see is "how is the config here overall" (discover the keys, see the values, etc.), and I felt not convenient to need to go step by step to have all the information.
So we may just add a `overview` (or `all` or…) command or an option to list or whatever. But I'm sure I'm needing this now, and I will in the future, and not only me. Because I really think from now on when connecting to a server for debugging that will be one of the first thing we'll do to have the big picture. And it's also something we may want to add on logs at some points.
I'm just giving a user feedback, and making a suggestion, but that suggestion can be discussed to end on something better at the end to have this "config overview".